framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3106,7 +3106,7 @@ public boolean validateTypeOf(Tree tree) {
             }
             break;
         default:
-            type = atypeFactory.getAnnotatedType(tree);
+        	type = atypeFactory.getDefaultedAnnotatedType(tree);
```

Comment:
If we assume that `@B MyClass` is not a valid type, then we have to assume that `@D MyClass` and `@C MyClass` are also not valid types because they are subtypes of `@B MyClass`.  Right?
 
````
void foo(@D MyClass dParam, @C MyClass cParam, boolean flag) {
    @A MyClass d = dParam;
    @A MyClass c = cParam;
    @A MyClass cond = flag ? bRefined : cRefined;
}
````
Currently, the checker would issue an error for `@D MyClass dParam`, `@C MyClass cParam`, both places `dParam` and `cParam` appear, and also  for the expression `flag ? bRefined : cRefined`.

>There is no explicit B in the program, so what would the developer do?

The developer would see the errors on `@D MyClass dParam` and `@C MyClass cParam` and correct the annotations.

> should type validation only validate the explicitly written annotations 

Does this mean that errors would only be issued at `@D MyClass dParam` and `@C MyClass cParam`?